generate-assets/src/lib.rs

```diff
@@ -511,36 +512,36 @@ fn get_bevy_version_from_manifest(
 /// If it was a bit more generic, this function could be called "find_first_intersect_in_sorted_iterators".
 /// Both dependencies and bevy_crates are assumed to be sorted (by key for dependencies, they are in this context),
 /// and we find the first element that intersect both of them using that knowledge.
-fn search_bevy_in_manifest_dependencies<'a>(
-    mut dependencies: std::collections::btree_map::Range<'a, String, cargo_toml::Dependency>,
+fn search_bevy_in_manifest_dependencies(
+    mut dependencies: std::collections::btree_map::Range<'_, String, cargo_toml::Dependency>,
     mut bevy_crates: std::slice::Iter<String>,
 ) -> Option<String> {
     let mut dependency = dependencies.next();
     let mut bevy_crate = bevy_crates.next();
 
-    while !dependency.is_none() && !bevy_crate.is_none() {
+    while dependency.is_some() && bevy_crate.is_some() {
         let dependency_name = dependency.unwrap().0;
         let bevy_crate_name = bevy_crate.unwrap();
 
-        if dependency_name < bevy_crate_name {
-            dependency = dependencies.next();
-        } else {
-            if dependency_name == bevy_crate_name {
+        match dependency_name.cmp(bevy_crate_name) {
```

Comment:
This was suggested in a lint, but I'm unsure if I like it as compared to the original. Thoughts?